### PR TITLE
Add IFM and DIFM model

### PR DESCRIPTION
**Pull request by zanshuxun**
Add IFM and DIFM model

### CODE REVIEW

File: deepctr/feature_column.py [hunk 1]
```diff
@@ -166,10 +166,10 @@ def get_linear_logit(features, feature_columns, units=1, use_bias=False, seed=10
         if len(linear_emb_list[i]) > 0 and len(dense_input_list) > 0:
             sparse_input = concat_func(linear_emb_list[i])
             dense_input = concat_func(dense_input_list)
-            linear_logit = Linear(l2_reg, mode=2, use_bias=use_bias, seed=seed)([sparse_input, dense_input])
+            linear_logit = Linear(l2_reg, mode=2, use_bias=use_bias, seed=seed)([sparse_input, dense_input], sparse_feat_refine_weight=sparse_feat_refine_weight)
```
Comment (workingloong) [inline deepctr/feature_column.py:169]: The line is too long.

File: deepctr/layers/interaction.py [hunk 2]
```diff
@@ -1406,3 +1409,22 @@ def get_config(self):
             'regularizer': self.regularizer
         })
         return config
+
+
+class IFMSoftmax(Layer):
+    # The special softmax operation in IFM.
+    # In tf1.4, multiplying a integer and a tensor in Lambda Layer will cause error in load_model().
+    # Hence, this operation are wrapped in this class.
```
Comment (workingloong) [inline deepctr/layers/interaction.py:1417]: "are" -> "is"

File: deepctr/layers/interaction.py [hunk 3]
```diff
@@ -1406,3 +1409,22 @@ def get_config(self):
...
+    def __init__(self, sparse_feat_num, **kwargs):
+        self.sparse_feat_num = sparse_feat_num
+        super(IFMSoftmax, self).__init__()
+
+    def call(self, inputs, **kwargs):
+        return self.sparse_feat_num * softmax(inputs, dim=1)
+
+    def get_config(self):
+        config = super(IFMSoftmax, self).get_config().copy()
+        config.update({
```
Comment (workingloong) [inline deepctr/layers/interaction.py:1427]: `config['sparse_feat_num'] = self.sparse_feat_num`?

Comment (zanshuxun) [reply deepctr/layers/interaction.py]: Thank you for your insightful suggestions. I've addressed these issues in the new version.

Comment (zanshuxun) [reply deepctr/layers/interaction.py]: Thank you for your insightful suggestions. I've addressed these issues in the new version: https://github.com/shenweichen/DeepCTR/commit/3048a1eec979835a066f6315c30fbd04cdf4cbbc

**Pull request merged**